applications/nrf5340_audio/src/modules/lc3_playback.c

```diff
@@ -0,0 +1,228 @@
+#include "lc3_playback.h"
+#include <stdint.h>
+#include <zephyr/sys/ring_buffer.h>
+#include "sd_card.h"
+#include "hw_codec.h"
+#include "sw_codec_lc3.h"
+#include "sw_codec_select.h"
+#include "pcm_stream_channel_modifier.h"
+#include "pcm_mix.h"
+#include "audio_i2s.h"
+#include "sw_codec_lc3.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(lc3_playback, 4); /* For draft PR feedback: going to change this line */
+
+#define RING_BUF_SIZE		1920 /* This can be modified */
+#define lc3_playback_STACK_SIZE 4096
+
+/*File structure of the LC3 encoded files*/
+struct lc3BinaryHdr_t {
+	uint16_t fileId;		/* Constant value, 0xCC1C */
+	uint16_t hdrSize;		/* Header size, 0x0012 */
+	uint16_t sampleRate_divided100; /* Sample frequency / 100 */
+	uint16_t bitRate_divided100;	/* Bit rate / 100 (total for all channels) */
+	uint16_t channels;		/* Number of channels */
+	uint16_t frameMs_times100;	/* Frame duration in ms * 100 */
+	uint16_t rfu;			/* RFU value */
+	uint16_t signalLen;		/* Number of samples in signal, 16 LSB */
+	uint16_t signalLenRed;		/* Number of samples in signal, 16 MSB (>> 16) */
+};
+
+static struct lc3BinaryHdr_t header;
+static size_t header_size = sizeof(header);
+static uint16_t lc3_frame_length;
+static uint16_t lc3_frames_num;
+static uint16_t pcm_mono_frame_size;
+static bool lc3_playback_active;
+
+RING_BUF_DECLARE(m_ringbuf_sound_data_lc3, RING_BUF_SIZE);
+K_SEM_DEFINE(m_sem_load_from_buf_lc3, 1, 1);
+K_MUTEX_DEFINE(mtx_ringbuf);
+
+static struct k_thread lc3_playback_thread_data;
+static k_tid_t lc3_playback_thread_id;
+
+K_THREAD_STACK_DEFINE(lc3_playback_thread_stack, lc3_playback_STACK_SIZE);
+
+bool lc3_playback_is_active(void)
+{
+	return lc3_playback_active;
+}
+
+int lc3_playback_buffer_set(uint8_t *buf, size_t size)
+{
+	uint16_t size_gotten;
+
+	k_mutex_lock(&mtx_ringbuf, K_FOREVER);
```

Comment:
Just curious, would this mutex mandatory? Would this cause issue if we remove it? What would this mutex going to prevent?

Reply:
I thought it might avoid potential problems with a concurrent write operation. Like the one in the lc3_playback_buffer_to_ringbuffer-function.